src/utils.c

```diff
@@ -677,6 +677,21 @@ file_mode(struct error *err, const char *path, mode_t *mode)
         return (0);
 }
 
+// file_mode_nofollow implements the same functionality as file_mode except that
+// in that case of a symlink, the file is not followed and the mode of the
+// original file is returned.
+int
+file_mode_nofollow(struct error *err, const char *path, mode_t *mode)
+{
+        struct stat s;
+        int rv;
+
+        if ((rv = xlstat(err, path, &s)) < 0)
```

Comment:
Is `rv` really needed here?
```suggestion
        if (xlstat(err, path, &s) < 0)
```

Reply:
No. It's not. I originally had the nested conditionals as follow-on statements, but refactored. Will update.